rules/CodingStyle/Rector/Stmt/NewlineAfterStatementRector.php

```diff
@@ -138,14 +138,19 @@ public function refactor(Node $node): ?array
             }
 
             /** @var Comment[] $comments */
-            $line = $comments[0]->getLine();
+            $line = $comments[0]->getStartLine();
             $rangeLine = $line - $endLine;
 
             if ($rangeLine > 1) {
                 return null;
             }
         }
 
+        // skip same line that cause infinite loop
+        if ($rangeLine === 0) {
+            return null;
+        }
```

Comment:
Given the following code:

```php
if ($node->expr instanceof FuncCall) {
    return ! $node->expr->isFirstClassCallable() && $this->isName($node->expr, 'array_filter') && isset($node->expr->args[0]);
}
```
it cause infinite loop during extracting `&&` operator to each stmt, this handle it.